libindex/controllerfactory.go

```diff
@@ -7,21 +7,19 @@ import (
 	"github.com/quay/claircore/internal/indexer/controller"
 	"github.com/quay/claircore/internal/indexer/fetcher"
 	"github.com/quay/claircore/internal/indexer/layerscanner"
-	dlpg "github.com/quay/claircore/pkg/distlock/postgres"
 )
 
 // ControllerFactory is a factory method to return a Controller during libindex runtime.
 type ControllerFactory func(_ context.Context, lib *Libindex, opts *Opts) (*controller.Controller, error)
 
 // controllerFactory is the default ControllerFactory
 func controllerFactory(ctx context.Context, lib *Libindex, opts *Opts) (*controller.Controller, error) {
-	sc := dlpg.NewLock(lib.db, opts.ScanLockRetry)
 	ft := fetcher.New(lib.client, opts.LayerFetchOpt)
 
 	// convert libindex.Opts to indexer.Opts
 	sOpts := &indexer.Opts{
 		Store:         lib.store,
-		ScanLock:      sc,
+		ScanLock:      lib.locks,
```

Comment:
Is this an issue? A single lock cannot be shared and must be created on controller construction. This is a fallout of our previous lock design, its a lock per go routine. 

I belelieve LibIndex needs to be configured with a handle to the pool, and a NewPool lock needs to be constructed by this factory, or else attempting to lock for the same manifest will cause an error.